contracts/timed_asset_lock_contract/program.py

```diff
@@ -79,18 +88,26 @@ def closeAccountTo(account: Expr) -> Expr:
         Approve(),
     )
 
+    # OnOptIn handles when a wallet request to opt into this smart contract. Only the
+    # wallet receiving (and sending) the funds can opt into this smart contract.
+    on_opt_in = Seq(
+        Assert(
+            # Only the original creator and receiver can opt into this smart contract.
+            Gtxn[1].sender() == App.globalGet(receiver_address_key),
+        ),
+        Approve(),
+    )
+
     # OnSetup handles setting up this freeze smart contract. Namely, it tells this smart
     # contract to opt into the asset it was created to hold. This smart contract must
     # hold enough Algo's to make this transaction.
     on_setup = Seq(
         Assert(
             And(
                 # The wallet triggering the setup must be the original creator and receiver.
-                Txn.sender() == App.globalGet(receiver_address_key),
+                Gtxn[2].sender() == App.globalGet(receiver_address_key),
                 # Sender must be opted in
-                App.optedIn(Txn.sender(), App.id()),
-                # This smart contract must be set up before the unlock timestamp.
-                Global.latest_timestamp() < App.globalGet(unlock_time_key),

```

Comment:
Same here

Reply:
see above....we should only care about checking timestamp when claiming